data_steward/validation/main.py

```diff
@@ -751,29 +752,103 @@ def list_submitted_bucket_items(folder_bucketitems):
+
+        # build timestamps
+        tmp = initial_date_time_object(file_meta)
+        created_date = datetime.datetime(year=tmp.year,
+                                         month=tmp.month,
+                                         day=tmp.day)
+        updated_date = updated_date_time_object(file_meta)
+
+        # only add to return if the file is less than 30 days and more than 5 minutes old.
+        if created_date < (created_base - object_retention_days):
+            logging.warn(
+                f'Filtering file {fname} as it > 30 days old ({(created_base - created_date).days} days old)'
+            )
+        elif updated_date > (now - object_minimum_age):
+            logging.warn(
+                f'Filtering file {fname} as it was last modified less than 5 minutes ago ({(now - updated_date).seconds} second(s) old)'
+            )
+        else:
+            logging.debug(f'Adding {fname} to files_list')
+            files_list.append(file_meta)
+
     return files_list
 
 
+def parse_date_time_object(gcs_object_metadata, key: str) -> datetime.datetime:
+    """
+    parse_date_time_object extracts a particular value from the provided metadata,
+    attempting to parse it as a datetime object
+    
+    :param gcs_object_metadata: object metadata dict
+    :param key: key of target date value in metadata
+    :return datetime.datetime:
+    """
+    return datetime.datetime.strptime(gcs_object_metadata[key], _GCS_DATE_FMT)
+
+
 def initial_date_time_object(gcs_object_metadata):
     """
     :param gcs_object_metadata: metadata as returned by list bucket
     :return: datetime object
     """
-    date_created = datetime.datetime.strptime(
-        gcs_object_metadata['timeCreated'], '%Y-%m-%dT%H:%M:%S.%fZ')
-    return date_created
+    return parse_date_time_object(gcs_object_metadata, 'timeCreated')
+
+
+def updated_date_time_object(gcs_object_metadata):
+    """
+    updated_date_time_object returns a datetime.datetime instance built
+    from the last time the metadata indicates the object was updated
+
+    :param gcs_object_metadata: metadata dict for a particular object
+    :return: datetime.datetime
+    """
+    return parse_date_time_object(gcs_object_metadata, 'updated')
+
+
+
+def _get_missing_required_files(submitted_bucket_items):
+    """
+    _get_missing_required_files returns a list of the names of the "required"
+    we expect to see in a given HPO upload set.
+
+    If the returned list is greater than 1, the consumer should assume the
+    targeted set is invalid or incomplete
+
+    :param submitted_bucket_items: list of string "file" names from bucket
+    :return: list of strings of missing require file names
+    """
+
+    # if the provided input is empty, state that all required files are missing
+    if submitted_bucket_items is None or len(submitted_bucket_items) is 0:
+        return common.AOU_REQUIRED_FILES
```

Comment:
I would also suggest using 
```suggestion
    if not submitted_bucket_items:
```
for this check too.  I don't think you actually care if the identity value of the `len()` is the same identity as `0` either.  For that length check I would recommend, `len(submitted_bucket_items) == 0`.

Reply:
Good point, I still conflate `is` and `==` in python.